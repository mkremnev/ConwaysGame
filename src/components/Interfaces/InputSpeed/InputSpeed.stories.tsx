import React from 'react';
import { InputSpeed } from './InputSpeed';
import { action } from '@storybook/addon-actions';
import { withKnobs, number } from '@storybook/addon-knobs';

// eslint-disable-next-line no-restricted-syntax
export default {
	title: 'Lesson 8 / Interfaces / InputSpeedChange',
	decorators: [withKnobs],
};

export const InputSpeedChange = ({ ...props }) => [
	<InputSpeed
		speedValue={number('Value', props.speedValue)}
		onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
			e.preventDefault();
			const target = e.target.value;
			props.speedValue = +target;
			action('Input change')(props.speedValue);
		}}
		key="interfaces"
	/>,
];
