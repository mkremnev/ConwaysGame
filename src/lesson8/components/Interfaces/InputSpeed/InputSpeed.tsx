import React, { FC } from 'react';
import { InputSpeedType } from '../../../types/Interfaces/InputSpeedType';

export const InputSpeed: FC<InputSpeedType> = ({ speedValue, onChange }) => {
	return (
		<input
			type="range"
			min="50"
			max="1000"
			step="50"
			value={speedValue}
			onChange={onChange}
		/>
	);
};
