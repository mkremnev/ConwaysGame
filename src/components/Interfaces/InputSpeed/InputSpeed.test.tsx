import React from 'react';
import { mount } from 'enzyme';
import { InputSpeed } from './InputSpeed';
import renderer from 'react-test-renderer';

describe('BtnClearBoard', () => {
	it('renders button clear board', () => {
		expect(
			renderer
				.create(<InputSpeed speedValue={500} onChange={jest.fn()} />)
				.toJSON(),
		).toMatchSnapshot();
	});
	it('renders button clear board', () => {
		expect(
			renderer
				.create(<InputSpeed speedValue={500} onChange={jest.fn()} />)
				.toJSON(),
		).toMatchSnapshot();
	});
	it('calls change callback on change input', () => {
		const onChange = jest.fn();
		const wrapper = mount(
			<InputSpeed speedValue={500} onChange={onChange} />,
		);
		wrapper.simulate('change');
		expect(onChange).toHaveBeenCalled();
	});
});
