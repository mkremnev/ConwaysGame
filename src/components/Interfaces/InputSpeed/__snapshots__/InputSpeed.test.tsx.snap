// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnClearBoard renders button clear board 1`] = `
Array [
  <input
    max="1000"
    min="50"
    name="speedValue"
    onChange={[MockFunction]}
    step="50"
    type="range"
    value={500}
  />,
  <label
    htmlFor="speedValue"
  >
    Изменить скорость
  </label>,
]
`;

exports[`BtnClearBoard renders button clear board 2`] = `
Array [
  <input
    max="1000"
    min="50"
    name="speedValue"
    onChange={[MockFunction]}
    step="50"
    type="range"
    value={500}
  />,
  <label
    htmlFor="speedValue"
  >
    Изменить скорость
  </label>,
]
`;
